source/io.c

```diff
@@ -107,6 +108,43 @@ PyObject *aws_py_io_host_resolver_new_default(PyObject *self, PyObject *args) {
     return PyCapsule_New(host_resolver, s_capsule_name_host_resolver, s_host_resolver_destructor);
 }
 
+static void s_server_bootstrap_destructor(PyObject *bootstrap_capsule) {
+
+    assert(PyCapsule_CheckExact(bootstrap_capsule));
+    struct aws_server_bootstrap *bootstrap = PyCapsule_GetPointer(bootstrap_capsule, s_capsule_name_server_bootstrap);
+    assert(bootstrap);
+    aws_server_bootstrap_release(bootstrap);
+}
+
+PyObject *aws_py_io_server_bootstrap_new(PyObject *self, PyObject *args) {
+    (void)self;
+
+    struct aws_allocator *allocator = aws_crt_python_get_allocator();
+
+    PyObject *elg_capsule = NULL;
+
+    if (!PyArg_ParseTuple(args, "O", &elg_capsule)) {
+        return NULL;
+    }
+
+    if (!elg_capsule || !PyCapsule_CheckExact(elg_capsule)) {
+        PyErr_SetNone(PyExc_ValueError);
+        return NULL;
```

Comment:
this if-branch can be deleted

I realize you were just copying the client_bootstrap code, but that was some of the earlier binding code written. We later found that PyCapsule_GetPointer() also checks for null and ensures the argument is a capsule. There's no need to repeat the checks